### ISSUE TITLE
Custom path separator characters

### ISSUE THREAD

**Issue by Tovkal** [joelittlejohn/jsonschema2pojo#688]
Hi,

I've got some jsonschemas from [Sabre](developer.sabre.com) and need to generate the Java classes for it, using the maven plugin. Here's the [request schema](http://files.developer.sabre.com/doc/providerdoc/STPS/bfm/v300/BargainFinderMax_v3.0.0_Request.jsonschema) for one of their APIs, [BargainFinderMax](https://developer.sabre.com/docs/read/rest_apis/air/search/bargain_finder_max). The problem I have with these schemas is that the definitions' names are all fully-qualified Java names.

A simplified version of the schema that reproduces the problem would be this:

```
{
  "$schema" : "http://json-schema.org/draft-04/schema#",
  "title" : "com.tovkal.FooRQ",
  "type" : "object",
  "properties" : {
    "FooRQ" : {
      "type" : "object",
      "properties" : {
        "POS" : {
          "$ref" : "#/definitions/com.tovkal.Bar"
        }
      }
    }
  },
  "definitions" : {
    "com.tovkal.Bar" : {
      "type" : "object",
      "properties" : {
        "AgentSine" : {
          "type" : "string"
        }
      }    }
  }
}
```

Running this schema using the maven plugin throws this error:

```
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.jsonschema2pojo:jsonschema2pojo-maven-plugin:0.4.30:generate (default) on project Testing: Execution default of goal org.jsonschema2pojo:jsonschema2pojo-maven-plugin:0.4.30:generate failed: Path not present: com
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:47)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal org.jsonschema2pojo:jsonschema2pojo-maven-plugin:0.4.30:generate failed: Path not present: com
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 26 more
Caused by: java.lang.IllegalArgumentException: Path not present: com
	at org.jsonschema2pojo.FragmentResolver.resolve(FragmentResolver.java:54)
	at org.jsonschema2pojo.FragmentResolver.resolve(FragmentResolver.java:52)
	at org.jsonschema2pojo.FragmentResolver.resolve(FragmentResolver.java:31)
	at org.jsonschema2pojo.SchemaStore.create(SchemaStore.java:51)
	at org.jsonschema2pojo.SchemaStore.create(SchemaStore.java:116)
	at org.jsonschema2pojo.rules.SchemaRule.apply(SchemaRule.java:59)
	at org.jsonschema2pojo.rules.SchemaRule.apply(SchemaRule.java:30)
	at org.jsonschema2pojo.rules.PropertyRule.apply(PropertyRule.java:75)
	at org.jsonschema2pojo.rules.PropertyRule.apply(PropertyRule.java:43)
	at org.jsonschema2pojo.rules.PropertiesRule.apply(PropertiesRule.java:64)
	at org.jsonschema2pojo.rules.PropertiesRule.apply(PropertiesRule.java:32)
	at org.jsonschema2pojo.rules.ObjectRule.apply(ObjectRule.java:115)
	at org.jsonschema2pojo.rules.ObjectRule.apply(ObjectRule.java:66)
	at org.jsonschema2pojo.rules.TypeRule.apply(TypeRule.java:90)
	at org.jsonschema2pojo.rules.TypeRule.apply(TypeRule.java:40)
	at org.jsonschema2pojo.rules.SchemaRule.apply(SchemaRule.java:73)
	at org.jsonschema2pojo.rules.SchemaRule.apply(SchemaRule.java:30)
	at org.jsonschema2pojo.rules.PropertyRule.apply(PropertyRule.java:75)
	at org.jsonschema2pojo.rules.PropertyRule.apply(PropertyRule.java:43)
	at org.jsonschema2pojo.rules.PropertiesRule.apply(PropertiesRule.java:64)
	at org.jsonschema2pojo.rules.PropertiesRule.apply(PropertiesRule.java:32)
	at org.jsonschema2pojo.rules.ObjectRule.apply(ObjectRule.java:115)
	at org.jsonschema2pojo.rules.ObjectRule.apply(ObjectRule.java:66)
	at org.jsonschema2pojo.rules.TypeRule.apply(TypeRule.java:90)
	at org.jsonschema2pojo.rules.TypeRule.apply(TypeRule.java:40)
	at org.jsonschema2pojo.rules.SchemaRule.apply(SchemaRule.java:73)
	at org.jsonschema2pojo.rules.SchemaRule.apply(SchemaRule.java:66)
	at org.jsonschema2pojo.rules.SchemaRule.apply(SchemaRule.java:30)
	at org.jsonschema2pojo.SchemaMapper.generate(SchemaMapper.java:92)
	at org.jsonschema2pojo.Jsonschema2Pojo.generateRecursive(Jsonschema2Pojo.java:109)
	at org.jsonschema2pojo.Jsonschema2Pojo.generate(Jsonschema2Pojo.java:73)
	at org.jsonschema2pojo.maven.Jsonschema2PojoMojo.execute(Jsonschema2PojoMojo.java:650)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	... 27 more
```

From the stack trace I got to the [FragmentResolver](https://github.com/joelittlejohn/jsonschema2pojo/blob/master/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/FragmentResolver.java) class and saw that on the first _resolve_ method it splits the path using `"#/."` as separator characters. 

I downloaded the source, modified the separator characters to `"#/"` and it generated the POJOs just fine.

So, is there an option that I might have missed that would allow me to generate the POJOs from these jsonschemas? Or is it possible to add an option to customize these separator characters? Should and could this option be added, a default value of `"#/."` should keep existing projects working.

**Comment by joelittlejohn** [comment]
I *think* the reason for including a dot here (this code is many years old and was written not long after the project was created IIRC) was to support JSON Path.

People *should* be using JSON Pointer, so it's unfortunate that this dot is present. There will be projects relying on the current behavior though (as you quite rightly point out) so yes, I think a configuration option to choose the list of separators would be good here. I don't see a reason not to add this.

**Comment by massfords** [comment]
FYI - the sample code from [Sabre](https://github.com/SabreDevStudio/SACS-Java#rest-1) mentions this plugin and how code gen from a schema file doesn't work that well. I was skeptical and glad I investigated. I think the biggest problem is this issue which I'm glad to see will get fixed. There are also other issues in their schema like dupe fields that fail code gen and some non-fatal errors like inconsistent types.

Like @Tovkal, I've been tweeting their schema manually with success.

**Comment by joelittlejohn** [comment]
Closed by #697
